delphi-frontend/src/main/java/au/com/integradev/delphi/type/intrinsic/IntrinsicReturnType.java

```diff
@@ -89,11 +93,37 @@ public static Type argumentByIndex(int index) {
     return new ArgumentByIndexReturnType(index);
   }
 
+  private static final class LengthReturnType extends IntrinsicReturnType {
+    private final Type byteType;
+    private final Type integerType;
+    private final Type nativeIntType;
+
+    private LengthReturnType(TypeFactory typeFactory) {
+      this.byteType = typeFactory.getIntrinsic(IntrinsicType.BYTE);
+      this.integerType = typeFactory.getIntrinsic(IntrinsicType.INTEGER);
+      this.nativeIntType = typeFactory.getIntrinsic(IntrinsicType.NATIVEINT);
+    }
+
+    @Override
+    public Type getReturnType(List<Type> arguments) {
```

Comment:
This is Delphi 12-only behaviour, but it's being applied for all versions. Is this intentional?

Reply:
Nope! Good catch.